Remove draft/metadata-notify-2 (#559)
ircv3/ircv3-specifications: extensions/metadata.md

On joining a server, clients have to configure their 'metadata key [subscriptions](#metadata-sub)'. This is a list of which keys the client understands and wants to get updates about (for example, they may subscribe to a `status` key if they support user-set statuses). By default, this subscription list is empty.

> When a channel's metadata is updated, all users in that channel who are subscribed to the changed key will receive a [`METADATA` message](#metadata-server-message) notifying them of the change. When a user's metadata is updated, all users either sharing a channel with that user or [monitoring](https://ircv3.net/specs/extensions/monitor.html#monitor-command) them will receive a `METADATA` message notifying them of the change.

Reviewer: ```suggestion
When metadata is updated, users who have subscribed to the changed key will receive a [`METADATA` message](#metadata-server-message) notifying them of the change.

For channel metadata, this applies only to subscribers who are members of the updated channel. For user metadata, this applies only to subscribers who either share a channel with the updated user or are [monitoring](https://ircv3.net/specs/extensions/monitor.html#monitor-command) them.
```

Clarify who gets metadata notifications.

This specification also uses the [standard replies](../extensions/standard-replies.html) framework.

> Clients MUST NOT request both (`metadata` or `metadata-notify`) and (`draft/metadata-2`). Servers MUST NOT accept these requests either.

Reviewer: The previous deprecated incompatible spec was just `metadata-notify` unprefixed. This spec is `draft/metadata-2` and will ideally end up as `metadata` unprefixed.

There is no such spec as `draft/metadata` or `draft/metadata-notify`.

> This specification replaces the first `metadata` specification, by adding the following incompatible changes:

Reviewer: ```suggestion
This specification replaces the `metadata-notify` cap defined in the deprecated [metadata-3.2](https://ircv3.net/specs/core/metadata-3.2) specification, by adding the following incompatible changes:
```

The heading of this section should also be updated, but it's not in this changeset so github won't let me add a suggestion.

Author: the link won't exist next time we update the website

Reviewer: ```suggestion
This specification replaces an earlier deprecated `metadata-notify` specification, by adding the following incompatible changes:
```

Reviewer: @progval happy with revised suggestion added above?

> * The `metadata-notify` key subscribed you to all keys. Since we have now added the [`SUB`](#metadata-sub) and [`UNSUB`](#metadata-unsub) subcommands, it is no longer deemed useful.

Reviewer: ```suggestion
* The `metadata-notify` cap subscribed you to all keys. With the addition of explicit [`SUB`](#metadata-sub) and [`UNSUB`](#metadata-unsub) subcommands, this is no longer the case.
```
